programs/Makefile

```diff
@@ -67,7 +67,10 @@ endif
 
 # zlib detection
 VOID = /dev/null
-HAVE_ZLIB := $(shell echo "\#include <zlib.h>\nint main(){}" | $(CC) -o $(VOID) -x c - -lz 2> $(VOID) && echo 1 || echo 0)
+HAVE_ZLIB := $(shell echo -e "\#include <zlib.h>\nint main(){}" | $(CC) -o have_zlib -x c - -lz 2> $(VOID) && echo 1 || echo 0)
+ifeq ($(HAVE_ZLIB), 1)
```

Comment:
But the issue here is that
```
ifeq ($(HAVE_ZLIB), 1)
TEMP := $(shell rm have_zlib$(EXT))
	@echo "==> building zstd with .gz decompression support "
else
	@echo "==> no zlib, building zstd with .zst support only (no .gz support) "
endif
```
breaks the `zstd-internal` target. I'm not sure what is your solution.